src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -33,6 +36,40 @@ internal static partial class PathInternal
         };
 
         /// <summary>
+        /// Returns true if the path is too long
+        /// </summary>
+        internal static bool IsPathTooLong(string fullPath)
+        {
+            if (IsExtended(fullPath))
+            {
+                return fullPath.Length >= MaxExtendedPath;
+            }
+            else
+            {
+                // Will need to be updated with #2581 to allow all paths to MaxExtendedPath
+                // minus legth of extended local or UNC prefix.
+                return fullPath.Length >= MaxShortPath;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the directory is too long
+        /// </summary>
+        internal static bool IsDirectoryTooLong(string fullPath)
+        {
+            if (IsExtended(fullPath))
+            {
+                return fullPath.Length >= MaxExtendedPath;
+            }
+            else
+            {
+                // Will need to be updated with #2581 to allow all paths to MaxExtendedPath
+                // minus legth of extended local or UNC prefix.
```

Comment:
Thanks- going to ignore this one for now as the change fixes a build issue. I'll be changing this comment when I add the implicit support.